Wikipedia/Code/InternalLinkPreviewing.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+protocol InternalLinkPreviewing: ViewController { }
+
+extension InternalLinkPreviewing {
+    func showInternalLink(url: URL) {
+        let exists: Bool
+        if let query = url.query {
+            exists = !query.contains("redlink=1")
+        } else {
+            exists = true
+        }
+        if !exists {
+            showRedLinkInAlert()
+            return
+        }
+        let dataStore = MWKDataStore.shared()
+        let internalLinkViewController = EditPreviewInternalLinkViewController(articleURL: url, dataStore: dataStore)
```

Comment:
If the article link doesn't exist (like this screenshot, to "https://en.wikipedia.org/wiki/List_of_Republicans_who_have_denounced_Donald_Trump" on "2020 United States presidental election"), the preview hangs indefinitely.

![loading](https://user-images.githubusercontent.com/5652327/98419077-228fb480-2039-11eb-8218-9e13be6b363f.png)

Reply:
This one probably should be filed separately. There are two failures here: 
1) We're looking for `redlink=1` in the url's query to decide whether to show this as a "no actual article" alert or this mini-article view. That is not present in [the data returned from the server](https://mobileapps-ios-experiments.wmflabs.org/en.wikipedia.org/v1/page/significant-events/2020_United_States_presidential_election?rvstartid=987405390). Maybe that's related to why [we're not handling redlinks more generally](https://phabricator.wikimedia.org/T267438), I'm not sure.
2) This bit of code, which formerly was just used for EditPreview, didn't actually do anything on a server error. I've done a quick and dirty stopping of the spinner, so you just get a white screen. But this should probably be improved to show some sort of error in this case.